src/util.c

```diff
@@ -162,20 +168,27 @@ struct passwd *hyper_getpwnam(const char *name)
 		struct passwd *pwd = fgetpwent(file);
 		if (!pwd)
 			break;
-		if (!strcmp(pwd->pw_name, name) || pwd->pw_uid == uid) {
+		if (pwd->pw_uid == uid) {
 			fclose(file);
 			return pwd;
 		}
+		if (!strcmp(pwd->pw_name, name)) {
+			user_pwd = pwd;
+		}
```

Comment:
please covert the logic to
if ((INVALID_UGID == uid && !strcmp(pwd->pw_name, name)) || pwd->pw_uid == uid) {

so that "10" is always uid instead name.

Reply:
updated , thanks.